libmozevent/mercurial.py

```diff
@@ -154,10 +154,19 @@ def apply_build(self, build):
         except hglib.error.CommandError:
             raise Exception("Failed to update to revision {}".format(hg_base))
 
-        # Get current revision using full information tuple from hglib
-        revision = self.repo.identify(id=True).strip()
-        revision = self.repo.log(revision, limit=1)[0]
-        logger.info("Updated repo", revision=revision.node, repo=self.name)
+        # In this case revision is `hg_base`
+        logger.info("Updated repo", revision=hg_base, repo=self.name)
+
+        # See if the repo is clean
+        repo_status = self.repo.status()
+        if len(repo_status) != 0:
+            status = ""
+            for elem in repo_status:
+                if elem[0] == 'C':
+                    continue
+                status += elem[1] + ": " + elem[0] + "\n"
+
+            logger.info("Repo is dirty! Modified files: \n{}".format(status), revision=hg_base, repo=self.name)
```

Comment:
Make `status` a parameter like `revision` or `repo` instead of adding it in the message

Reply:
This shouldn't happen, so rather than `info`, let's use `warn`